src/harness/sas_interface.py

```diff
@@ -476,3 +476,34 @@ def assertRegisteredAndGranted(self, registration_request, grant_request,
       A list of  dictionaries, each with keys "cbsdId" and "grantId".
     """
     pass
+
+  @abc.abstractmethod
+  def assertHeartbeatsSuccessful(self, cbsd_ids, grant_ids, operation_states):
+    """Make a heartbeat requests for the list of devices with its grants and 
+    operation states.
+    
+    Sends heartbeat requests and assert the response for valid cbsd id,
+     grant id, and transmit expire time
+    Args:
+      cbsd_ids: A list containing cbsd Ids.
+      grant_ids: A list containing grant Id associated with each device in 
+      cbsd_ids list.
+      operation_states: A list containing operation states (AUTHORIZED or 
+      GRANTED) for each devices in the cbsd_ids list.
+    Returns:
+      A list of transmit expire time in the format YYYY-MM-DDThh:mm:ssZ.    
+    """
+    pass
+
+  @abc.abstractmethod
+  def TriggerDailyActivitiesImmediatelyAndWaitUntilComplete(self):
```

Comment:
I added this method, TriggerDailyActivitiesImmediatelyAndWaitUntilComplete in the same PR. May be I will create another PR for this because this was an issue where I forgot to add this method in the sas_interface.py as we agreed to shift this helper method to sas_testcase.py.